ragna/assistants/_ollama.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import AsyncIterator, cast
+
+from ragna.core import RagnaException, Source
+
+from ._api import UnauthenticatedApiAssistant
+
+
+class OllamaApiAssistant(UnauthenticatedApiAssistant):
+    _MODEL: str
+
+    @classmethod
+    def display_name(cls) -> str:
+        return f"Ollama/{cls._MODEL}"
+
+    def _make_system_content(self, sources: list[Source]) -> str:
+        instruction = (
+            "You are an helpful assistants that answers user questions given the context below. "
+            "If you don't know the answer, just say so. Don't try to make up an answer. "
+            "Only use the following sources to generate the answer."
+        )
+        return instruction + "\n\n".join(source.content for source in sources)
+
+    async def _call_api(
+        self,
+        prompt: str,
+        sources: list[Source],
+        *,
+        max_new_tokens: int,
+        api_url: str = "http://localhost:11434/api/chat",
```

Comment:
This is not a runtime parameter, given that this URL doesn't change. Let's put it into `__init__`. We should also have a way to set this through an environment variable, e.g.

```python
def __init__(self, url="http://localhost:11434/api/chat"):
    self._url = os.environ.get("RAGNA_ASSISTANTS_OLLAMA_URL", url)
```